src/memory/mod.rs

```diff
@@ -10,11 +10,11 @@ use core::{
 };
 use parity_wasm::elements::ResizableLimits;
 
-#[cfg(all(unix, not(feature = "vec_memory")))]
+#[cfg(all(unix, feature = "virtual_memory"))]
 #[path = "mmap_bytebuf.rs"]
 mod bytebuf;
 
-#[cfg(any(not(unix), feature = "vec_memory"))]
+#[cfg(not(all(unix, feature = "virtual_memory")))]
```

Comment:
I don't like the automatic fallback but I guess we need to live with it because you can't have target depended features and it would make it annoying for downstream crates if we don't had the fallback. Maybe we can have some compile time warning if this fallback is happening?

Reply:
This should improve once we support more than just "unix" platforms for virtual memory.

Reply:
Resolved by https://github.com/paritytech/wasmi/pull/274